app/models/dynamic_field_group.rb

```diff
@@ -29,6 +29,11 @@ class DynamicFieldGroup < ActiveRecord::Base
   validate  :non_circular_relationship
   validates_with StringKey::AbsentInSiblingFieldsValidator
 
+  def save
```

Comment:
@barmintor I had another idea for how we could handle HYACINTH-568.  I'm going to open a PR to this branch in a sec so you can check it out and see what you think.  It involves two extra SQL queries, but since we don't update field definitions too often I'm hoping it doesn't seem too bad, and it allows us to avoid overriding `save`.